### ISSUE TITLE
tmp folder grows to huge size, making the docker container huge

### ISSUE THREAD

**Issue by pewter77** [allanpk716/ChineseSubFinder#54]
Apologies that this is in English, I download the subs for my GF so we have dual language subtitles. This program works wonders, but my technical Chinese is not very good.

## 你使用的 chinesesubfinder 是什么版本，什么环境？unraid (docker-compose)

chinesesubfinder 版本: v0.16.0

环境: docker

## 你遇到什么问题了？

/tmp folder becomes very large making the docker container 9GB+

## 你的问题弄重现嘛？

Yes

## 你期望的结果

I know that chrome is used to scrappers to and browser emulators to download the subtitles due to no API, but the tmp folder probably needs to be cleaned every once in a while.

## 可以提供遇到问题时候，电影、连续剧的信息吗？

N/A

## 出错的日志信息可以复制一些出来吗？

I can get the logs but not sure they're relevant right now.

**Comment by allanpk716** [comment]
Can you run the following command inside the container to see what is returned?

`du -h --max-depth=1 /tmp`

## pics
![image](https://user-images.githubusercontent.com/1176000/134317867-c206b29c-dfe5-4a1b-8aa3-9d1a26b60bcb.png)

**Comment by pewter77** [comment]
This is only recent, though the /tmp/rod size is huge.
I recreated the container just this morning.

0       /tmp/.X11-unix
13M     /tmp/chinesesubfinder
2.3M    /tmp/leakless-34a05a9dc363ec03e25d5dcc5ff915d2
1.8G    /tmp/rod
4.0K    /tmp/.org.chromium.Chromium.D1joIh
4.0K    /tmp/.org.chromium.Chromium.jm0x4F
4.0K    /tmp/.org.chromium.Chromium.ZTwgTf
4.0K    /tmp/.org.chromium.Chromium.vMRqEc
4.0K    /tmp/.org.chromium.Chromium.tuFzeE
4.0K    /tmp/.org.chromium.Chromium.HCUajb
4.0K    /tmp/.org.chromium.Chromium.GsOmeT
4.0K    /tmp/.org.chromium.Chromium.6Af1KW
4.0K    /tmp/.org.chromium.Chromium.kUCV22
4.0K    /tmp/.org.chromium.Chromium.OpMrtv
4.0K    /tmp/.org.chromium.Chromium.YIUlYz
4.0K    /tmp/.org.chromium.Chromium.9bCqUA
4.0K    /tmp/.org.chromium.Chromium.eYxxv0
4.0K    /tmp/.org.chromium.Chromium.IRhCQB
4.0K    /tmp/.org.chromium.Chromium.WEaE6c
4.0K    /tmp/.org.chromium.Chromium.6aTtCM
4.0K    /tmp/.org.chromium.Chromium.G0qv1n
4.0K    /tmp/.org.chromium.Chromium.Mr8wfl
4.0K    /tmp/.org.chromium.Chromium.4h0L8X
4.0K    /tmp/.org.chromium.Chromium.HO9uMn
4.0K    /tmp/.org.chromium.Chromium.8fzAmT
4.0K    /tmp/.org.chromium.Chromium.4EeQcE
4.0K    /tmp/.org.chromium.Chromium.zg3TdA
4.0K    /tmp/.org.chromium.Chromium.b1qEOp
4.0K    /tmp/.org.chromium.Chromium.Aby59S
4.0K    /tmp/.org.chromium.Chromium.yFHdMJ
4.0K    /tmp/.org.chromium.Chromium.eWXs92
4.0K    /tmp/.org.chromium.Chromium.Cu8Iqc
4.0K    /tmp/.org.chromium.Chromium.DUv7gP
4.0K    /tmp/.org.chromium.Chromium.iv1Vy6
4.0K    /tmp/.org.chromium.Chromium.dggzzr
2.6M    /tmp/.org.chromium.Chromium.kElG0f
4.0K    /tmp/.org.chromium.Chromium.EOoBtP
4.0K    /tmp/.org.chromium.Chromium.7ieTiz
1.8G    /tmp

**Comment by allanpk716** [comment]
Apparently rod's cache is taking up a lot of space. Does the program(ChineseSubFinder) run all day long(7 x 24h) in your use?

Can you run the following command inside the container to see what is returned?

`du -h --max-depth=1 /tmp/rod`

`ls /tmp/rod/user-data`

`ls /tmp/rod/downloads`

`du -h --max-depth=1 /tmp/rod/user-data`

`du -h --max-depth=1 /tmp/rod/downloads`

**Comment by pewter77** [comment]
I do run it daily, 6h intervals.

```
UseProxy: false
HttpProxy: http://127.0.0.1:10809
EveryTime: 6h
Threads: 2
SubTypePriority: 0
SubNameFormatter : 1
DebugMode: false 
SaveMultiSub: true
RunAtStartup: true
MovieFolder: /data/media/movies
SeriesFolder: /data/media/tv
AnimeFolder: /data/media/anime

EmbyConfig :
    Url : http://emby:8096 
    ApiKey : <removed> 
    LimitCount : 5000
    SkipWatched : false
```

```
**du -h --max-depth=1 /tmp/rod**
2.0G    /tmp/rod/user-data
0       /tmp/rod/downloads
2.0G    /tmp/rod
```
```
**ls /tmp/rod/user-data**
0aa72fe1f6e36f7c  1a1924bdc86642b7  346e7784348b7773  559f18e47bfaf2f9  615596f986860631  8afa7565134c9919  9e9e170e877f9c0c  b664cba564ac42b7  d481b9b87b4c1bc2  ed842a0590f59daa
0bfab7426feaddda  1de1a549c20d0077  34b685b678478e06  562f70804c26cf66  6f8306394ad1a6a0  8ba560f4256624fa  9f5bce5893fcf6de  bb7c201a5e8f72e8  d56dcd95ea2b0398  edb8f56635636934
0c075e08bbe4d3e3  1e2d9ee38a909c8c  34d507ecc76b7244  563c19f1a09bf8bf  72f08cc136e964a7  8e3d7daff7766323  a3a44bd16d2cc6c0  bc03172c3c0784b1  d684bcc265911b6f  ee0e1e998ff24d4f
0d49c17d5d8b5bb2  2092d80305868c7f  35dd1b6209473ea7  5a23e065e2429330  7583d4945075420d  90ceba048adb5adf  a3ae87036989cc6c  beaec25b31a6c96d  da49386bbf43eaa6  f0fb6d32392201b3
118cd06263d05457  295ce853e0a0fe34  3946c8792f781dbc  5b0def5cb6707237  77391e964294cb5b  920c937c4132bef7  af60b35dcfb60559  c54f336f357ef387  df0a0345ba0e2873  f38ca2375776ce71
16388d00f4f581fb  2a7cac900065524c  3c33bad52ea1ce6e  5ba7a2429e208d36  7c4dab07bc50afbc  93372f5c107e353d  b1c8f175374fc62e  c950ebbb43e81367  e1d0c1274707c2b3  f3f699f5012533cf
1804cd18cde353e7  2d0c93e61561ef1f  3cfc2cb95e40d575  5d72eb5918277d83  807e1952dccaa429  9646d7656fdcf248  b32f6fef71ab08af  d30904ca80d5b27c  e644c55d0f916b4f  f9ad53a7f9ed30f4
1952d951dd340f4a  2f573eaa2a875a89  41ee09c05532e026  5d926430b2352a9c  866ebfeac91d510d  978c8972494654d0  b3d9653cece62f91  d3267fd1caa7deaf  ecf00380a6c5c100  ff991ee073f5dbf8
```

`**ls /tmp/rod/downloads**`

```
**du -h --max-depth=1 /tmp/rod/user-data**
11M     /tmp/rod/user-data/e644c55d0f916b4f
18M     /tmp/rod/user-data/d3267fd1caa7deaf
18M     /tmp/rod/user-data/c950ebbb43e81367
7.2M    /tmp/rod/user-data/ff991ee073f5dbf8
7.0M    /tmp/rod/user-data/41ee09c05532e026
6.4M    /tmp/rod/user-data/1804cd18cde353e7
6.3M    /tmp/rod/user-data/da49386bbf43eaa6
6.7M    /tmp/rod/user-data/b1c8f175374fc62e
19M     /tmp/rod/user-data/3cfc2cb95e40d575
64M     /tmp/rod/user-data/77391e964294cb5b
18M     /tmp/rod/user-data/e1d0c1274707c2b3
19M     /tmp/rod/user-data/72f08cc136e964a7
33M     /tmp/rod/user-data/16388d00f4f581fb
5.9M    /tmp/rod/user-data/ee0e1e998ff24d4f
66M     /tmp/rod/user-data/f9ad53a7f9ed30f4
54M     /tmp/rod/user-data/0bfab7426feaddda
6.9M    /tmp/rod/user-data/2f573eaa2a875a89
41M     /tmp/rod/user-data/1a1924bdc86642b7
7.5M    /tmp/rod/user-data/5d926430b2352a9c
2.9M    /tmp/rod/user-data/0d49c17d5d8b5bb2
5.8M    /tmp/rod/user-data/8afa7565134c9919
3.2M    /tmp/rod/user-data/563c19f1a09bf8bf
2.9M    /tmp/rod/user-data/93372f5c107e353d
2.9M    /tmp/rod/user-data/f0fb6d32392201b3
8.0M    /tmp/rod/user-data/34d507ecc76b7244
6.7M    /tmp/rod/user-data/0aa72fe1f6e36f7c
2.8M    /tmp/rod/user-data/346e7784348b7773
55M     /tmp/rod/user-data/9e9e170e877f9c0c
28M     /tmp/rod/user-data/bb7c201a5e8f72e8
2.8M    /tmp/rod/user-data/0c075e08bbe4d3e3
61M     /tmp/rod/user-data/c54f336f357ef387
18M     /tmp/rod/user-data/b664cba564ac42b7
18M     /tmp/rod/user-data/35dd1b6209473ea7
18M     /tmp/rod/user-data/2a7cac900065524c
23M     /tmp/rod/user-data/beaec25b31a6c96d
23M     /tmp/rod/user-data/559f18e47bfaf2f9
23M     /tmp/rod/user-data/bc03172c3c0784b1
5.9M    /tmp/rod/user-data/ed842a0590f59daa
23M     /tmp/rod/user-data/f38ca2375776ce71
23M     /tmp/rod/user-data/5a23e065e2429330
23M     /tmp/rod/user-data/1e2d9ee38a909c8c
18M     /tmp/rod/user-data/edb8f56635636934
61M     /tmp/rod/user-data/f3f699f5012533cf
18M     /tmp/rod/user-data/3c33bad52ea1ce6e
18M     /tmp/rod/user-data/8ba560f4256624fa
18M     /tmp/rod/user-data/5ba7a2429e208d36
23M     /tmp/rod/user-data/8e3d7daff7766323
23M     /tmp/rod/user-data/615596f986860631
23M     /tmp/rod/user-data/807e1952dccaa429
23M     /tmp/rod/user-data/978c8972494654d0
61M     /tmp/rod/user-data/5d72eb5918277d83
55M     /tmp/rod/user-data/2092d80305868c7f
62M     /tmp/rod/user-data/a3a44bd16d2cc6c0
49M     /tmp/rod/user-data/7c4dab07bc50afbc
49M     /tmp/rod/user-data/1952d951dd340f4a
18M     /tmp/rod/user-data/118cd06263d05457
20M     /tmp/rod/user-data/d30904ca80d5b27c
5.9M    /tmp/rod/user-data/d684bcc265911b6f
18M     /tmp/rod/user-data/b32f6fef71ab08af
18M     /tmp/rod/user-data/90ceba048adb5adf
23M     /tmp/rod/user-data/a3ae87036989cc6c
23M     /tmp/rod/user-data/562f70804c26cf66
5.9M    /tmp/rod/user-data/ecf00380a6c5c100
23M     /tmp/rod/user-data/3946c8792f781dbc
23M     /tmp/rod/user-data/2d0c93e61561ef1f
81M     /tmp/rod/user-data/af60b35dcfb60559
83M     /tmp/rod/user-data/34b685b678478e06
70M     /tmp/rod/user-data/920c937c4132bef7
70M     /tmp/rod/user-data/295ce853e0a0fe34
24M     /tmp/rod/user-data/d56dcd95ea2b0398
6.4M    /tmp/rod/user-data/1de1a549c20d0077
18M     /tmp/rod/user-data/9646d7656fdcf248
19M     /tmp/rod/user-data/9f5bce5893fcf6de
2.8M    /tmp/rod/user-data/7583d4945075420d
18M     /tmp/rod/user-data/df0a0345ba0e2873
70M     /tmp/rod/user-data/6f8306394ad1a6a0
70M     /tmp/rod/user-data/b3d9653cece62f91
8.1M    /tmp/rod/user-data/866ebfeac91d510d
67M     /tmp/rod/user-data/5b0def5cb6707237
71M     /tmp/rod/user-data/d481b9b87b4c1bc2
2.1G    /tmp/rod/user-data
```

```
**du -h --max-depth=1 /tmp/rod/downloads**
0       /tmp/rod/downloads
```

**Comment by allanpk716** [comment]
like I said,  /tmp/rod is the [go-rod](https://github.com/go-rod/rod) cache, and I find the code in go-rod, see [this](https://github.com/go-rod/rod/blob/52c3253bb115a8882854bca34387f165491f05e7/lib/launcher/launcher.go).

## pics
![image](https://user-images.githubusercontent.com/1176000/134462246-54e9ecec-1f0e-4d50-9d82-bfa54a12194a.png)

## Options under consideration

I can set up to clean this folder after the task is completed, but this is strongly related to an external code logic. 

I need to rethink my options.

**Comment by wloot** [comment]
> like I said, /tmp/rod is the [go-rod](https://github.com/go-rod/rod) cache, and I find the code in go-rod, see [this](https://github.com/go-rod/rod/blob/52c3253bb115a8882854bca34387f165491f05e7/lib/launcher/launcher.go).
> 
> ## pics
> ![image](https://user-images.githubusercontent.com/1176000/134462246-54e9ecec-1f0e-4d50-9d82-bfa54a12194a.png)
> 
> ## Options under consideration
> I can set up to clean this folder after the task is completed, but this is strongly related to an external code logic.
> 
> I need to rethink my options.

rod有清理该缓存文件夹的接口 https://github.com/go-rod/rod/blob/a7722e5f40bd66b1119028fd1c7353a28395dc85/examples_test.go#L71 , 在执行任务后调用下就好了.

**Comment by allanpk716** [comment]
> > like I said, /tmp/rod is the [go-rod](https://github.com/go-rod/rod) cache, and I find the code in go-rod, see [this](https://github.com/go-rod/rod/blob/52c3253bb115a8882854bca34387f165491f05e7/lib/launcher/launcher.go).
> > ## pics
> > ![image](https://user-images.githubusercontent.com/1176000/134462246-54e9ecec-1f0e-4d50-9d82-bfa54a12194a.png)
> > ## Options under consideration
> > I can set up to clean this folder after the task is completed, but this is strongly related to an external code logic.
> > I need to rethink my options.
> 
> rod有清理该缓存文件夹的接口 https://github.com/go-rod/rod/blob/a7722e5f40bd66b1119028fd1c7353a28395dc85/examples_test.go#L71 , 在执行任务后调用下就好了.

感谢提供方案。

**Comment by allanpk716** [comment]
fix in v0.17.3
